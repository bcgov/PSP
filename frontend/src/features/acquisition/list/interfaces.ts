export interface IAcquisitionFilter {
  acquisitionFileStatusTypeCode: string;
  acquisitionFileNameOrNumber: string;
  projectNameOrNumber: string;
}
