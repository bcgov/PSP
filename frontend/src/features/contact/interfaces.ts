export interface IContactFilter {
  summary: string;
  municipality: string;
  activeContactsOnly: boolean;
  searchBy: string;
}
