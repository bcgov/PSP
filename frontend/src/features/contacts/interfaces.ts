export interface IContactFilter {
  summary: string;
  municipality: string;
  activeContactsOnly: boolean;
  searchBy: string;
}

export enum ContactTypes {
  ORGANIZATION = 'O',
  INDIVIDUAL = 'P',
}
