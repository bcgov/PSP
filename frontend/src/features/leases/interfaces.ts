export interface ILeaseFilter {
  pinOrPid: string;
  lFileNo: string;
  searchBy: string;
  tenantName: string;
  programs: string[];
}
