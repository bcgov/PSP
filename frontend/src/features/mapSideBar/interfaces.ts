import { AddressTypes, PropertyTypes } from 'constants/index';
import { IAddress } from 'interfaces';

export const defaultPropertyValues = {
  address: {
    addressTypeId: AddressTypes.Physical,
    streetAddress1: '',
    municipality: '',
    postal: '',
    province: '',
    provinceId: 1,
    region: '',
    district: '',
  } as IAddress,
  pid: '',
  propertyTypeId: PropertyTypes.Land,
  classification: '',
  region: '',
  district: '',
  zoning: '',
  landArea: '',
  landLegalDescription: '',

  // TODO: Theses properties probably should be replaced or updated.
  titleNumber: '',
  purpose: '',
  ruralArea: '',
  ocpDesignation: '',
};
