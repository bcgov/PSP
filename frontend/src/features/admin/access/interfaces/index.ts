import { AccessRequestStatus } from 'constants/accessStatus';

export interface IAccessRequestModel {
  id: number;
  userId: number;
  businessIdentifierValue: string;
  firstName: string;
  surname: string;
  email: string;
  position: string;
  status: AccessRequestStatus;
  organization: string;
  role: string;
  note: string;
}
