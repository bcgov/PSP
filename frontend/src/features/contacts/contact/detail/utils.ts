import { IContactAddress } from 'interfaces/IContact';

import { AddressTypes } from './../../../../constants/addressTypes';

// the order of this array corresponds to the expected display order
const addressSortOrder = [AddressTypes.Mailing, AddressTypes.Residential, AddressTypes.Billing];

export const sortAddresses = (a1: IContactAddress, a2: IContactAddress) => {
  if (a2.addressType.id === a1.addressType.id) {
    return 0;
  }
  const a2Index = addressSortOrder.indexOf(a2.addressType.id as AddressTypes);
  if (a2Index !== -1 && a2Index < addressSortOrder.indexOf(a1.addressType.id as AddressTypes)) {
    return 1;
  }

  return -1;
};

export const fakeAddresses = [
  {
    id: 1,
    rowVersion: 3,
    addressType: {
      id: 'BILLING',
      description: 'Billing address',
      isDisabled: false,
    },
    streetAddress1: 'Billing address',
    municipality: 'Hollywood North',
    province: {
      provinceStateId: 1,
      provinceStateCode: 'BC',
      description: 'British Columbia',
    },
    country: {
      countryId: 1,
      countryCode: 'CA',
      description: 'Canada',
    },
    postal: 'V6A 5G7',
  },
  {
    id: 1,
    rowVersion: 3,
    addressType: {
      id: 'MAILING',
      description: 'Mailing address',
      isDisabled: false,
    },
    streetAddress1: 'Mailing address',
    streetAddress2: 'Living in a van',
    streetAddress3: 'Down by the River',
    municipality: 'Hollywood North',
    province: {
      provinceStateId: 1,
      provinceStateCode: 'BC',
      description: 'British Columbia',
    },
    country: {
      countryId: 1,
      countryCode: 'CA',
      description: 'Canada',
    },
    postal: 'V6Z 5G7',
  },
  {
    id: 5,
    rowVersion: 1,
    addressType: {
      id: 'RESIDNT',
      description: 'Property address',
      isDisabled: false,
    },
    streetAddress1: 'Property address',
    streetAddress2: '',
    streetAddress3: '',
    municipality: 'vic',
    province: {
      provinceStateId: 1,
      provinceStateCode: 'BC',
      description: 'British Columbia',
    },
    country: {
      countryId: 1,
      countryCode: 'CA',
      description: 'Canada',
    },
    postal: 'V9A 0H6',
  },
];
