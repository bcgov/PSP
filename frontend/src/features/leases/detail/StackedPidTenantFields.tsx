import OverflowTip from 'components/common/OverflowTip';
import { InlineFlexDiv } from 'components/common/styles';
import { ILease } from 'interfaces/ILease';
import * as React from 'react';
import styled from 'styled-components';
import { pidFormatter } from 'utils';

import { getAllNames } from '../leaseUtils';
export interface IStackedPidTenantFieldsProps {
  lease?: ILease;
}

/**
 * Layout component that displays vertically stacked lease pids and tenant names
 * @param {IStackedPidTenantFieldsProps} param0
 */
export const StackedPidTenantFields: React.FunctionComponent<IStackedPidTenantFieldsProps> = ({
  lease,
}) => {
  const properties = lease?.properties ?? [];
  const pids = properties.map(property => pidFormatter(property.pid));
  return (
    <>
      <StyledStackedDivs>
        <label>PID:</label>
        <OverflowTip title="pids" fullText={pids.join(', ')} />
      </StyledStackedDivs>
      <StyledStackedDivs>
        <label>Tenant:</label>
        <OverflowTip title="tenant" fullText={getAllNames(lease)} />
      </StyledStackedDivs>
    </>
  );
};

const StyledStackedDivs = styled(InlineFlexDiv)`
  width: 100%;
  text-align: left;
  div {
    font-weight: 700;
  }
`;

export default StackedPidTenantFields;
