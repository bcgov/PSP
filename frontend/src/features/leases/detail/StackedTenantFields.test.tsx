import { createMemoryHistory } from 'history';
import { defaultLease } from 'interfaces';
import { render, RenderOptions } from 'utils/test-utils';

import StackedPidTenantFields, { IStackedTenantFieldsProps } from './StackedTenantFields';

const history = createMemoryHistory();
const onClickManagement = jest.fn();

describe('StackedPidTenantFields component', () => {
  const setup = (
    renderOptions: RenderOptions & IStackedTenantFieldsProps = { lease: { ...defaultLease } },
  ) => {
    // render component under test
    const component = render(<StackedPidTenantFields lease={renderOptions.lease} />, {
      ...renderOptions,
      history,
    });

    return {
      component,
    };
  };
  beforeEach(() => {
    onClickManagement.mockReset();
  });
  it('renders as expected when provided no data', () => {
    const { component } = setup();
    expect(component.asFragment()).toMatchSnapshot();
  });
  it('renders as expected when provided data', () => {
    const { component } = setup({
      lease: {
        ...defaultLease,
        properties: [{ pid: '111-111-111' } as any],
        persons: [{ firstName: 'First', surname: 'Last' }],
      },
    });
    expect(component.asFragment()).toMatchSnapshot();
  });

  it('renders tenant appropriately', () => {
    const {
      component: { getByText },
    } = setup({
      lease: { ...defaultLease, persons: [{ firstName: 'tenantFirst', surname: 'tenantSurname' }] },
    });
    expect(getByText('tenantFirst tenantSurname')).toBeVisible();
  });
});
