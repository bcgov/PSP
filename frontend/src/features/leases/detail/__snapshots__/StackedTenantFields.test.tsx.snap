// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StackedPidTenantFields component renders as expected when provided data 1`] = `
<DocumentFragment>
  @media only screen and (max-width:1250px) {

}

<div
    class="Toastify"
  />
  @media only screen and (max-width:1250px) {

}

<div
    class="Toastify"
  />
  .c2 {
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
}

.c1 {
  width: 100%;
  text-align: left;
}

.c1 div {
  font-family: 'BCSans-Bold';
}

@media only screen and (max-width:1250px) {
  .c1 {
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
  }
}

<div
    class="c0 c1"
  >
    <label>
      Tenant:
    </label>
    <div
      class="c2"
    >
      First Last
    </div>
  </div>
</DocumentFragment>
`;

exports[`StackedPidTenantFields component renders as expected when provided no data 1`] = `
<DocumentFragment>
  @media only screen and (max-width:1250px) {

}

<div
    class="Toastify"
  />
  @media only screen and (max-width:1250px) {

}

<div
    class="Toastify"
  />
  .c2 {
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
}

.c1 {
  width: 100%;
  text-align: left;
}

.c1 div {
  font-family: 'BCSans-Bold';
}

@media only screen and (max-width:1250px) {
  .c1 {
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
  }
}

<div
    class="c0 c1"
  >
    <label>
      Tenant:
    </label>
    <div
      class="c2"
    />
  </div>
</DocumentFragment>
`;
